vpr/src/route/bucket.cpp

```diff
@@ -141,10 +142,15 @@ void Bucket::free_all_memory() {
 void Bucket::expand(size_t required_number_of_buckets) {
     auto old_size = heap_size_;
     heap_size_ = required_number_of_buckets * 2;
-
-    heap_ = (BucketItem**)vtr::realloc((void*)(heap_),
-                                       heap_size_ * sizeof(BucketItem*));
-    std::fill(heap_ + old_size, heap_ + heap_size_, nullptr);
+    int i;
+
+    std::vector<BucketItem*> temp(heap_, heap_ + old_size);
+    delete[] heap_;
+    heap_ = new BucketItem*[heap_size_];
+    for (i = 0; i < (int)old_size; i++)
+        heap_[i] = temp[i];
```

Comment:
Same comment: better to do this without the temp vector.  Just allocate heap_new, copy over from heap_, delete heap_, then set heap_ = heap_new.